@here/harp-datasource-protocol/lib/Expr.ts

```diff
@@ -443,8 +458,12 @@ export class NumberLiteralExpr extends LiteralExpr {
  * @hidden
  */
 export class StringLiteralExpr extends LiteralExpr {
+    readonly constantValue?: RGBA | Pixels;
+
     constructor(readonly value: string) {
         super();
+
+        this.constantValue = RGBA.parse(value) ?? Pixels.parse(value);
```

Comment:
I don't think it's good to parse everything already in constructor. Good part of string literals processed by  engine are not actually RBGA/Pixel values but feature attribute names and enum-like literals.

`constantValue` is also weird name, i think it's `parsedValue` or `promotedValue`

I would write it as 
```
get promotedValue() {
     if not already tried to parse:
         this.m_promoted = this.m_promoted ?? RGBA.parse(value) ?? Pixels.parse(value);
     return this.m_promoted
}
```

Reply:
fixed